### PR TITLE
Blob storage

### PR DESCRIPTION
**Pull request by msohn**
**Pull Request checklist**

- [x ] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?
- [x ] Tests for the changes have been added (for bug fixes / features) ?
existing tests have good test coverage for the new code
- [x ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (existing issues : #876)

On H2 and Postgresql BYTEA column type is used to store binaries via byte array in Scala code

**New behavior :**

Use column type BLOB on H2 and OID on Postgresql to store binaries using the large object type supported by the respective database platform. Use JDBC BLOB API for managing large object content.
In the Scala code binaries are still passed as byte arrays. This should be improved in a future change to use streaming in order to reduce memory consumption.

Data is migrated to the new column type if necessary (on H2 and Postgresql).

**Other information**:

### CODE REVIEW

**Comment by msohn** [comment]
Can someone can answer my question on the mailing list how to setup local Python test environment needed to run the Python based tests ?
https://groups.google.com/forum/#!topic/spark-jobserver/pWN-KsCv_Rc

File: README.md [hunk 1]
```diff
@@ -555,43 +557,45 @@ Log files are separated out for each context (assuming `context-per-jvm` is `tru
...
 
-      jdbc {
-        url = "jdbc:postgresql://db_host/spark_jobserver"
-        user = "secret"
-        password = "secret"
-      }
+          # Directory where default H2 driver stores its data. Only needed for H2.
+          rootdir = "/var/spark-jobserver/sqldao/data"
+
+          jdbc {
```
Comment (velvia) [inline README.md]: I wonder if the database config part of doc is getting too big and we should move them to doc/.  I suppose that can be a separate PR.

File: job-server/src/main/scala/db/h2/migration/V0_7_2/V0_7_2__Migrate_Blobs.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  def migrate(c: Connection): Unit = {
+    val renameBinaryToBinArray = sqlu"""ALTER TABLE BINARIES ALTER COLUMN BINARY RENAME TO BINARRAY"""
+    val addBlobColumn = sqlu"""ALTER TABLE BINARIES ADD COLUMN BINARY BLOB"""
+    val dropBinArray = sqlu"""ALTER TABLE BINARIES DROP COLUMN BINARRAY"""
+    implicit val getBinArrayResult = GetResult[BinArray](r => BinArray(r.nextInt(), r.nextBytes()))
+    val getBinArrays = sql"""SELECT BIN_ID, BINARRAY FROM BINARIES""".as[BinArray]
+
+    val db = new UnmanagedDatabase(c)
+    c.setAutoCommit(false)
+    Await.ready(db.run(DBIO.seq(renameBinaryToBinArray, addBlobColumn)).recover{logErrors}, Timeout)
```
Comment (velvia) [inline job-server/src/main/scala/db/h2/migration/V0_7_2/V0_7_2__Migrate_Blobs.scala]: You could use a for comprehension to sequence all these Futures from db.run, then just use a single Await.ready and recover.

File: README.md (hunk 1 above)
Comment (msohn) [inline README.md]: I can move this to another document under doc/ in another PR after my vacation. I'll be back beginning of August

**Comment by velvia** [comment]
Sounds good, have a great vacation!

File: job-server/src/main/scala/db/h2/migration/V0_7_2/V0_7_2__Migrate_Blobs.scala (hunk 2 above)
Comment (msohn) [inline job-server/src/main/scala/db/h2/migration/V0_7_2/V0_7_2__Migrate_Blobs.scala]: done in c849a88700314d7b3e56982f9bf6d3f597aebef2

**Comment by msohn** [comment]
rebased this pull request to resolve schema migration version conflict with d536063874de60eac6383469f006d5a4ddd13e1d which was merged recently

**Comment by msohn** [comment]
The codecov report claims that this PR reduces coverage for LongPiJob but I can't see how this could happen.

Any hints ?

**Comment by thoHeinze** [comment]
@velvia Can we merge this change ?

**Comment by velvia** [comment]
Done  :)

**Comment by msohn** [comment]
thanks :-)
